### PR TITLE
feat(data-export-plugin): some improvements

### PR DESCRIPTION
**Pull request by Wroud**
dbeaver-corp/dbeaver-web#618
CB-102

### CODE REVIEW

File: webapp/packages/core/src/app/shared/NavigationTabs/NavigationTabsService.ts [hunk 1]
```diff
@@ -47,8 +47,8 @@ export class NavigationTabsService {
     return this.state.tabs;
   }
 
-  onTabSelect = new Subject<ITab>();
-  onTabClose = new Subject<ITab>();
+  readonly onTabSelect = new Subject<ITab>();
+  readonly onTabClose = new Subject<ITab>();
```
Comment (dkrupenya) [inline webapp/packages/core/src/app/shared/NavigationTabs/NavigationTabsService.ts]: the correct version is 
```
private _onTabSelect = new Subject<ITab>();
readonly onTabSelect = this._onTabSelect.asObservable()

File: webapp/packages/core/src/utils/Promises/Deferred.ts [hunk 2]
```diff
@@ -74,9 +74,10 @@ export class Deferred<T> {
   }
 
   @action
-  protected toCancelled(): void {
+  protected toCancelled(reason?: any): void {
     this.state = EDeferredState.CANCELLED;
-    this.promiseExecutor.reject(new PromiseCancelledError());
+    this.rejectionReason = reason;
+    this.promiseExecutor.reject(reason || new PromiseCancelledError());
```
Comment (dkrupenya) [inline webapp/packages/core/src/utils/Promises/Deferred.ts]: strictly declined.
pass reason as param to PromiseCancelledError()

File: webapp/packages/core/src/utils/Promises/Deferred.ts [hunk 3]
```diff
@@ -101,7 +102,7 @@ export class DeferredFromPromise<T> extends Deferred<T> {
       value => this.toResolved(value),
       (err) => {
         if (err instanceof PromiseCancelledError) {
-          this.toCancelled();
+          this.toCancelled(err);
```
Comment (dkrupenya) [inline webapp/packages/core/src/utils/Promises/Deferred.ts]: take reason from PromiseCancelledError and pass it to .toCancelled

File: webapp/packages/data-export-plugin/src/ExportNotification/Controller.ts [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+import { IInitializableController, injectable } from '@dbeaver/core/di';
+import { CommonDialogService } from '@dbeaver/core/dialogs';
+import { INotification } from '@dbeaver/core/eventsLog';
+import { LocalizationService } from '@dbeaver/core/localization';
+import { EDeferredState } from '@dbeaver/core/utils';
+
+import { DataExportProcessService } from '../DataExportProcessService';
+
+@injectable()
+export class Controller implements IInitializableController {
```
Comment (dkrupenya) [inline webapp/packages/data-export-plugin/src/ExportNotification/Controller.ts]: ExportNotificationController

**Review by dkrupenya — CHANGES_REQUESTED** [review]

File: webapp/packages/data-export-plugin/src/ExportNotification/ExportNotification.tsx [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+    setMounted(true);
+  }, []);
+
+  return styled(useStyles(SNACKBAR_STYLES, styles))(
+    <notification as="div" {...use({ mounted })} >
+      <notification-header as="div">
+        <Loader loading={controller.isPending} hideMessage />
+        <message as="div">{translate(controller.status)}</message>
+        {!controller.isPending && (
+          <ReakitButton onClick={controller.delete}>
```
Comment (dkrupenya) [inline webapp/packages/data-export-plugin/src/ExportNotification/ExportNotification.tsx]: Create separate companent in @dbeaver/core/blocks
Plugins should know nothing about reakit

**Review by dkrupenya — CHANGES_REQUESTED** [review]

File: webapp/packages/data-export-plugin/src/ExportNotification/ExportNotification.tsx (hunk 5 above)
Comment (Wroud) [reply webapp/packages/data-export-plugin/src/ExportNotification/ExportNotification.tsx]: IconButton implemented

File: webapp/packages/core/src/app/shared/NavigationTabs/NavigationTabsService.ts (hunk 1 above)
Comment (Wroud) [reply webapp/packages/core/src/app/shared/NavigationTabs/NavigationTabsService.ts]: fixed

File: webapp/packages/core/src/utils/Promises/Deferred.ts (hunk 3 above)
Comment (Wroud) [reply webapp/packages/core/src/utils/Promises/Deferred.ts]: fixed

File: webapp/packages/core/src/utils/Promises/Deferred.ts (hunk 2 above)
Comment (Wroud) [reply webapp/packages/core/src/utils/Promises/Deferred.ts]: fixed

File: webapp/packages/data-export-plugin/src/ExportNotification/Controller.ts (hunk 4 above)
Comment (Wroud) [reply webapp/packages/data-export-plugin/src/ExportNotification/Controller.ts]: fixed

**Review by dkrupenya — APPROVED** [review]

**Pull request merged**